addons/sale_stock/controllers/portal.py

```diff
@@ -36,3 +36,20 @@ def portal_my_picking_report(self, picking_id, access_token=None, **kw):
             ('Content-Length', len(pdf)),
         ]
         return request.make_response(pdf, headers=pdfhttpheaders)
+
+    @route(['/my/picking/return/pdf/<int:picking_id>'], type='http', auth="public", website=True)
+    def portal_my_picking_return_report(self, picking_id, access_token=None, **kw):
+        """ Print return label for customer, using either access rights or access token
+        to be sure customer has access """
+        try:
+            picking_sudo = self._stock_picking_check_access(picking_id, access_token=access_token)
+        except exceptions.AccessError:
+            return request.redirect('/my')
```

Comment:
I would return a notfound instead of the redirect so it hides the existence of the page. Maybe it's excessive.

Reply:
Also it tracaback if you use an id that does not exists

Reply:
I see the portal_my_picking_report function just above it has the same issues, would you like me to patch these as well?